### PR TITLE
Enrich all the tweets!

### PR DESCRIPTION
**Pull request by chris-hinds**
Resolves N/A

**Overall change:**
Adds a useEffect to lazyloaded embeds so we ca call `twttr.widgets.load()` which instructs the twitter JS to scan the dom again and enrich newly rendered tweets.

Example asset where secondary tweets and not "enriched": www.bbc.com/russian/features-54391793

Following @chris-hinds ' initial impelmentation I've built on this with:
- Seperate `EnrichTweet` component designed specifically to wrap a `CanonicalSocialEmbed` configured for twitter in a use effect
  - This component is only used for twitter content, other content uses an unwrapped `CanonicalSocialEmbed` component
- Add E2E test for enriching tweets after lazy load; a new test asset has been added to local testing and the live test has been adjusted to now target the same asset
- There are no unit tests for `EnrichTweet` on this PR these can follow on a seperate PR to expidite this bug fix to live; I judged the E2E gave adequate coverage but please feel free to challenge this

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [x] I have assigned this PR to the Simorgh project

**Testing:**

- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [x] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

**Review by rhenshaw56 — APPROVED** [review]

File: cypress/integration/pages/storyPage/testsForCanonicalOnly.js [hunk 1]
```diff
@@ -17,18 +17,50 @@ export const testsThatAlwaysRunForCanonicalOnly = ({ service }) => {
...
-  service,
-  pageType,
-}) =>
-  describe(`No testsThatFollowSmokeTestConfigForCanonicalOnly for ${service} ${pageType}`, () => {});
 
-// For testing low priority things e.g. cosmetic differences, and a safe place to put slow tests.
-export const testsThatNeverRunDuringSmokeTestingForCanonicalOnly = () => {
   describe('Social Embed', () => {
+    // This test specifically covers an edge case where more than one tweet is
+    // included in a Story and twitter need to prompted to render the tweet
```
Comment (chris-hinds) [inline cypress/integration/pages/storyPage/testsForCanonicalOnly.js]:
```suggestion
    // included in a Story and twitter need to be prompted to render the tweet
```

**Review by chris-hinds — COMMENTED** [review]
I cant approve as I authored the first part of the PR. Looks good to me though 👍

File: src/app/containers/SocialEmbed/index.jsx [hunk 2]
```diff
@@ -72,6 +73,24 @@ const SocialEmbedContainer = ({ blocks }) => {
...
+      caption={caption}
+    />
+  );
+
+  const enrichedSocialEmbed =
+    provider === 'twitter' ? (
+      <EnrichTweet>{socialEmbed}</EnrichTweet>
+    ) : (
+      socialEmbed
+    );
```
Comment (simonsinclair) [inline src/app/containers/SocialEmbed/index.jsx:92]: - This is called enrichedSocialEmbed, but could return an 'unenriched' social embed.
- This shouldn't be necessary, because our hook checks whether `twttr` is on the global object and it will only be there if there's a Social Embed from Twitter on the page.

File: src/app/containers/SocialEmbed/enrichTweet.jsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { useEffect } from 'react';
+import { shape } from 'prop-types';
+import socialEmbedBlockPropTypes from '#models/propTypes/socialEmbed';
+
+const EnrichTweet = ({ children }) => {
```
Comment (simonsinclair) [inline src/app/containers/SocialEmbed/enrichTweet.jsx:5]: Is there a need for this to be in a separate module? The three line callback within the useEffect is the only logic needed to load the widget after it's been rendered, and this code isn't and won't be shared.

File: src/app/containers/SocialEmbed/enrichTweet.test.jsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { render, waitFor } from '@testing-library/react';
+import EnrichTweet from './enrichTweet';
+import withContexts from './testHelper';
+import { twitterBlock } from './fixtures';
+import SocialEmbedContainer from '.';
+
+describe('EnrichTweet', () => {
+  it('should call twitter api to enrich tweets after initial render', async () => {
```
Comment (simonsinclair) [inline src/app/containers/SocialEmbed/enrichTweet.test.jsx:9]: This is an awesome test. 🤩

**Review by simonsinclair — COMMENTED** [review]
Nice work, Andrew. I've left a few thoughts below.

File: src/app/containers/SocialEmbed/index.jsx (hunk 2 above)
Comment (andrewscfc) [reply src/app/containers/SocialEmbed/index.jsx:92]: I'm happy to remove the conditional, it does though add an uneccessary `useEffect` to all social embeds that is avoided with this check. I don't know if adding lots of `useEffects` can harm the performance of the page; I imagine it's negligible?

File: src/app/containers/SocialEmbed/enrichTweet.jsx (hunk 3 above)
Comment (andrewscfc) [reply src/app/containers/SocialEmbed/enrichTweet.jsx:5]: Are you suggesting moving it to the index.jsx instead @simonsinclair ?

**Comment by paruchurisilpa** [comment]
Checked this using http://localhost:7080/russian/features-54391793 and the 2nd tweet LGTM now..

File: src/app/containers/SocialEmbed/index.jsx (hunk 2 above)
Comment (simonsinclair) [reply src/app/containers/SocialEmbed/index.jsx:92]: Yes, it's more what's running in the callback as opposed to the useEffect itself. The if-statement (inside the useEffect) guards against executing Twitter's logic unless it's needed.

File: src/app/containers/SocialEmbed/enrichTweet.jsx (hunk 3 above)
Comment (simonsinclair) [reply src/app/containers/SocialEmbed/enrichTweet.jsx:5]: I would suggest moving it to the SocialEmbedContainer, yeah. In my opinion, this should be call `twittr.load()` on render. It would make sense to separate this useEffect if it were shared.

**Comment by chris-hinds** [comment]
If we are happy with the logic in this PR can I suggest that we merge it now to minimise the live issues with social embeds and then fix the "stylistic" issues in a new PR

**Review by simonsinclair — APPROVED** [review]
